src/Server/Repositories/ClaraJobRepository.cs

```diff
@@ -0,0 +1,236 @@
+            }
+
+            throw new OperationCanceledException("Cancellation requested.");
+        }
+
+        private async Task UpdateInferenceJob(InferenceJob request)
+        {
+            Guard.Against.Null(request, nameof(request));
+
+            await Policy
+                 .Handle<Exception>()
+                 .WaitAndRetryAsync(
+                     3,
+                     retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
+                     (exception, timeSpan, retryCount, context) =>
+                     {
+                         _logger.Log(LogLevel.Warning, exception, $"Failed to update job. Waiting {timeSpan} before next retry. Retry attempt {retryCount}.");
+                     })
+                 .ExecuteAsync(async () =>
+                 {
+                     _logger.Log(LogLevel.Debug, $"Updating inference job.");
+                     await _inferenceJobRepository.SaveChangesAsync();
+                     _logger.Log(LogLevel.Debug, $"Inference job updated.");
+                 })
+                 .ConfigureAwait(false);
+        }
+
+        private InferenceJob CreateInferenceJob(Job job, string jobName, IList<InstanceStorageInfo> instances)
+        {
+            Guard.Against.Null(job, nameof(job));
+            Guard.Against.Null(jobName, nameof(jobName));
+            Guard.Against.Null(instances, nameof(instances));
+
+            var targetStoragePath = string.Empty; ;
+            if (_fileSystem.Directory.TryGenerateDirectory(_fileSystem.Path.Combine(_configuration.Value.Storage.Temporary, "jobs", $"{job.JobId}"), out targetStoragePath))
+            {
+                _logger.Log(LogLevel.Information, $"Job payloads directory set to {targetStoragePath}");
+                return new InferenceJob(targetStoragePath, job) { Instances = instances };
+            }
+            else
+            {
+                throw new JobStoreException($"Failed to generate a temporary storage location");
+            }
+        }
+
+        private void MakeACopyOfPayload(InferenceJob request)
+        {
+            _logger.Log(LogLevel.Information, $"Copying {request.Instances.Count} instances to {request.JobPayloadsStoragePath}.");
+            var files = new Stack<InstanceStorageInfo>(request.Instances);
+            var retrySleepMs = 1000;
+            var retryCount = 0;
+
+            while (files.Count > 0)
+            {
+                try
+                {
+                    var file = files.Peek();
+                    var destPath = _fileSystem.Path.Combine(request.JobPayloadsStoragePath, $"{file.SopInstanceUid}.dcm");
+                    _fileSystem.File.Copy(file.InstanceStorageFullPath, destPath, true);
+                    _logger.Log(LogLevel.Debug, $"Instance {file.SopInstanceUid} moved to {destPath}");
+                    files.Pop();
+                }
+                catch (IOException ex) when ((ex.HResult & 0xFFFF) == 0x27 || (ex.HResult & 0xFFFF) == 0x70)
```

Comment:
magic numbers are usually bad, can we name `0x27` and `0x70`?

Reply:
Fixed 🙏